dm/dm/master/shardddl/optimist.go

```diff
@@ -595,9 +595,30 @@ func (o *Optimist) handleSourceTables(ctx context.Context, sourceCh <-chan optim
 			if !ok {
 				return
 			}
-			updated := o.tk.Update(st)
-			o.logger.Info("receive source tables", zap.Stringer("source tables", st),
-				zap.Bool("is deleted", st.IsDeleted), zap.Bool("updated", updated))
+			o.mu.Lock()
+			addedTable, droppedTable := o.tk.Update(st)
+			// handle create table
+			for routeTable := range addedTable {
+				lock := o.lk.FindLock(utils.GenDDLLockID(st.Task, routeTable.DownSchema, routeTable.DownTable))
```

Comment:
do we need to create a lock for it when not found a lock?

Reply:
no need, we only create a lock when receive ddl which changed table schema.